grails-app/taglib/org/pih/warehouse/SelectTagLib.groovy

```diff
@@ -719,6 +737,22 @@ class SelectTagLib {
                 writer << '</optgroup>'
                 writer.println()
             }
+            noGroup.each {
```

Comment:
Wow, what is going on here. Besides "handling null group case".

Could you also add some comments to explain this?